### PR TITLE
Terminal refactoring

### PR DESCRIPTION
**Pull request by hexa00**
The goal is to make the terminal reusable with other services like the upcoming debug terminal

### CODE REVIEW

**Comment by hexa00** [comment]
Note I plan to create abstract functions for that is in the constructor rather than just overriding it as a whole...

**Comment by hexa00** [comment]
This is no longer a WIP, I've tested this with my debug console branch and it works properly so the refactoring should be ok for that need.

Please review.

**Comment by hexa00** [comment]
Still missing a bit there... working on it

**Comment by hexa00** [comment]
Please review, more refactoring will come but will be part of the debug patch series

File: src/terminal/browser/terminal-frontend-contribution.ts [hunk 1]
```diff
@@ -21,6 +21,8 @@ export namespace TerminalCommands {
     }
 }
 
+let terminalNum = 0
```
Comment (akosyakov) [inline src/terminal/browser/terminal-frontend-contribution.ts]: Should not it be a state of `TerminalFrontendContribution`?

Comment (hexa00) [inline src/terminal/browser/terminal-frontend-contribution.ts]: oops right I forgot to put it there, will fix

**Pull request merged**